### DISCUSSION TITLE
Sentence piece unigrams vs BPE for citrinet-like models

### DISCUSSION THREAD

**Discussion by jprobichaud** [NVIDIA-NeMo/NeMo#2129] [General]
I'm curious what people think makes more sense: for CTC-based models, like Citrinet, would sentence piece BPE tokenizer be better than unigram tokenizers?

I have some evidence that suggests that unigram tokenizers would fly well with attention decoders but make the life of a CTC decoder much harder.  

Anyone as notice that too?  Any potential justification for this?

**Comment by titu1994** [comment]
Good question ! We use WPE for Librispeech models, but unigram for most of the Nemo checkpoint releases. We did experiment with SPE BPE, and found no significant difference in WER for Librispeech compared to either unigram or WPE. However these weren't extensive analysis so there is a margin for noise. 

Overall, it would be interesting to know if SPE unigram does better than BPE for a particular dataset / case.

**Reply by jprobichaud** [reply]
Thanks.  We are still trying to understand the impact of all this given different architectures on our data.  I'll try to remember to update this thread once we'll have some usable conclusions.
